packages/color-scale-block/src/ColorScaleBlock.spec.ct.tsx

```diff
@@ -0,0 +1,641 @@
+            editorState: true,
+            blockSettings: { customHeight: false, heightSlider: '96px', heightInput: '100px' },
+        });
+
+        const COLOR_PICKER_FLYOUT_COLOR_INDEX_1 = 0;
+        const COLOR_PICKER_FLYOUT_COLOR_INDEX_2 = 1;
+
+        mount(<ColorScaleBlockWithStubs />);
+
+        cy.on('uncaught:exception', (err) => {
+            if (err.message.includes('Cannot call hover while not dragging')) {
+                // return false to prevent the error from
+                // failing this test
+                return false;
+            }
+
+            // fail test on any other errors
+            return true;
+        });
+
+        cy.get(ColorScaleBlockEditorModeButtonsSelector).find(AddColorButtonSelector).contains('Add Color').realClick();
+
+        cy.get(ColorPickerFlyoutSelector)
+            .find(ColorPickerFlyoutBrandColorSelector)
+            .eq(COLOR_PICKER_FLYOUT_COLOR_INDEX_1)
+            .realClick();
+
+        cy.get(ColorPickerFlyoutSelector)
+            .parentsUntil('[role="dialog"]')
+            .find('[data-test-id="button-text"]')
+            .contains('Confirm')
+            .realClick()
+            .wait(500);
+
+        cy.get(ColorScaleBlockEditorModeButtonsSelector).find(AddColorButtonSelector).contains('Add Color').realClick();
+
+        cy.get(ColorPickerFlyoutSelector)
+            .find(ColorPickerFlyoutBrandColorSelector)
+            .eq(COLOR_PICKER_FLYOUT_COLOR_INDEX_2)
+            .realClick();
+
+        cy.get(ColorPickerFlyoutSelector)
+            .parentsUntil('[role="dialog"]')
+            .find('[data-test-id="button-text"]')
+            .contains('Confirm')
+            .realClick();
+
+        cy.get(ColorScaleBlockSelector)
+            .find(ColorSquareSelector)
+            .eq(COLOR_PICKER_FLYOUT_COLOR_INDEX_2)
+            .find(ColorSquareBackgroundSelector)
+            .invoke('attr', 'style')
+            .as('dragged_element_style');
+
+        cy.get(ColorScaleBlockSelector)
+            .find(ColorSquareSelector)
+            .eq(COLOR_PICKER_FLYOUT_COLOR_INDEX_2)
+            .find(ColorSquareBackgroundSelector)
+            .eq(0)
+            .realHover()
+            .wait(500)
+            .drag(`${ColorSquareDropZoneSelector}:eq(0)`, { force: true });
+
+        cy.wait(1000);
+
+        cy.get(ColorScaleBlockSelector)
+            .find(ColorSquareSelector)
+            .eq(COLOR_PICKER_FLYOUT_COLOR_INDEX_1)
+            .find(ColorSquareBackgroundSelector)
+            .invoke('attr', 'style')
+            .as('first_color_in_block_style');
+
+        cy.get('@dragged_element_style').then((dragged_element_style) => {
+            cy.get('@first_color_in_block_style').then((first_color_in_block_style) => {
+                expect(dragged_element_style).to.equal(first_color_in_block_style);
+            });
+        });
+    });
+
+    it('adds a color square which is 16px wide, starting from the 3rd color added', () => {
```

Comment:
I was able to update most of the tests so that colors are pre-populated. However, for the test that verifies if the 3rd color is 18px wide (the 2nd to last test), it requires that the pre-populated color squares have a width that is 50% of the block. Since I don't know how wide those squares should be, I am adding those squares manually within the test.

Silly question but is it possible to update the block settings after the block has been mounted?

Reply:
I found a way around the issue in my last comment by doing this. Now all tests have colors pre-populated (if needed).

`       cy.get('body')
            .eq(0)
            .then(($body) => {
                const pageWidth = $body[0].getBoundingClientRect().width;

                const [ColorScaleBlockWithStubs] = withAppBridgeBlockStubs(ColorScaleBlock, {
                    editorState: true,
                    blockSettings: {
                        customHeight: false,
                        heightSlider: '96px',
                        heightInput: '100px',
                        colorInput: [
                            {
                                red: 0,
                                green: 0,
                                blue: 255,
                                id: 2,
                                width: pageWidth / 2 - COLOR_SCALE_BLOCK_BORDER_WIDTH - COLOR_SCALE_BLOCK_PADDING,
                            },
                            {
                                red: 255,
                                green: 0,
                                blue: 0,
                                id: 1,
                                width: pageWidth / 2 - COLOR_SCALE_BLOCK_BORDER_WIDTH - COLOR_SCALE_BLOCK_PADDING,
                            },
                        ],
                    },
                });`